pkg/controller/replica.go

```diff
@@ -108,13 +109,48 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 			if !isCleanupRequested(azVolumeAttachment) {
 				go r.handleReplicaDelete(context.Background(), azVolumeAttachment)
 			}
-			// if replica attachment failed, delete it and create a replacement
 		} else if azVolumeAttachment.Status.State == azdiskv1beta2.AttachmentFailed {
-			// if attachment failed for replica AzVolumeAttachment, delete the CRI so that replace replica AzVolumeAttachment can be created.
-			if err := r.cachedClient.Delete(ctx, azVolumeAttachment); err != nil {
-				return reconcile.Result{Requeue: true}, err
+			// if the failed attachment is retriable, reset the state to pending
+			if azureutils.MapContains(azVolumeAttachment.Status.Annotations, consts.VolumeAttachRetryAnnotation) {
+				retryCount := 0
+				if value, exists := azureutils.GetFromMap(azVolumeAttachment.Status.Annotations, consts.VolumeAttachRetryCount); exists {
+					if n, err := strconv.Atoi(value); err == nil {
+						retryCount = n
+					} else {
+						r.logger.Error(err, "failed to convert retry count to int")
+					}
+				}
+
+				var updateFunc func(obj client.Object) error
+				if retryCount < r.config.ControllerConfig.VolumeAttachRetryLimit {
+					updateFunc = func(obj client.Object) error {
+						azva := obj.(*azdiskv1beta2.AzVolumeAttachment)
+						azureutils.RemoveFromMap(azva.Status.Annotations, consts.VolumeAttachRetryAnnotation)
+						azureutils.AddToMap(azva.Status.Annotations, consts.VolumeAttachRetryCount, strconv.Itoa(retryCount+1))
+						_, derr := updateState(azva, azdiskv1beta2.AttachmentPending, normalUpdate)
```

Comment:
Use `forceUpdate` mode for updating state from `AttachmentFailed` to `AttachmentPending`